users/views.py

```diff
@@ -164,4 +171,279 @@ def get(self, request, user_id, token):
                 status=status.HTTP_500_INTERNAL_SERVER_ERROR
-            )
\ No newline at end of file
+            )
+
+
+User = get_user_model()
+def get_default_role():
+    return UserRole.objects.get(role="user")
+
+class OAuthTokenObtainPairView(TokenObtainPairView):
+    """
+    Extended token authentication endpoint that supports both traditional email/password
+    and OAuth provider authentication (Google/GitHub).
+    
+    Inherits from Djoser's TokenObtainPairView to maintain all standard functionality
+    while adding OAuth support through a unified authentication endpoint.
+    
+    Endpoint: users/oauth/login/
+    Methods: POST
+    
+    Request Formats:
+        - OAuth: {"provider": "google|github", "access_token": "oauth_token"}
+        - Password: {"email": "user@example.com", "password": "password123"}
+    """
+    
+    def post(self, request, *args, **kwargs):
+        """
+        Handle authentication requests, routing to appropriate method based on input.
+        
+        Args:
+            request: DRF request object containing authentication credentials
+            *args: Additional positional arguments
+            **kwargs: Additional keyword arguments
+            
+        Returns:
+            Response: JSON containing either:
+                - JWT tokens and user data (success)
+                - Error message (failure)
+                
+        Status Codes:
+            - 200 OK: Successful authentication
+            - 400 Bad Request: Missing/invalid parameters
+            - 401 Unauthorized: Invalid credentials
+        """
+        # Check if this is an OAuth request
+        provider = request.data.get('provider')  # 'google' or 'github'
+        access_token = request.data.get('access_token')
+        
+        if provider and access_token:
+            return self.handle_oauth(provider, access_token)
+        
+        # Fall back to standard email/password auth
+        return super().post(request, *args, **kwargs)
+    
+    def handle_oauth(self, provider, access_token):
+        """
+        Route OAuth authentication requests to the appropriate provider handler.
+        
+        Args:
+            provider: String identifying the OAuth provider ('google' or 'github')
+            access_token: Valid OAuth access token from the provider
+            
+        Returns:
+            Response: JSON response from the provider handler or error if unsupported
+            
+        Raises:
+            HTTP 400: If provider is not supported
+        """
+        if provider == 'google':
+            return self.handle_google_oauth(access_token)
+        elif provider == 'github':
+            return self.handle_github_oauth(access_token)
+        else:
+            return Response(
+                {"error": "Unsupported OAuth provider"}, 
+                status=status.HTTP_400_BAD_REQUEST
+            )
```

Comment:
Normalize provider string (e.g., .lower().strip()) for robustness against user input.